pkg/action/release_testing.go

```diff
@@ -108,20 +106,16 @@ func (r *ReleaseTesting) GetPodLogs(out io.Writer, rel *release.Release) error {
 	}
 
 	for _, h := range rel.Hooks {
-		for _, e := range h.Events {
-			if e == release.HookTest {
-				req := client.CoreV1().Pods(r.Namespace).GetLogs(h.Name, &v1.PodLogOptions{})
-				logReader, err := req.Stream(context.Background())
-				if err != nil {
-					return errors.Wrapf(err, "unable to get pod logs for %s", h.Name)
-				}
-
-				fmt.Fprintf(out, "POD LOGS: %s\n", h.Name)
-				_, err = io.Copy(out, logReader)
-				fmt.Fprintln(out)
-				if err != nil {
-					return errors.Wrapf(err, "unable to write pod logs for %s", h.Name)
-				}
+		if isTestHook(h) {
+			hookLog, err := getHookLog(client, rel, h)
+			if err != nil {
+				return err
+			}
+			fmt.Fprintf(out, "POD LOGS: %s\n", h.Name)
+			_, err = fmt.Fprint(out, string(hookLog))
+			fmt.Fprintln(out)
```

Comment:
This will `panic` if `out` cannot be written to. `err` needs to be handled before you attempt to call `Fprint` a second time.

I don't understand why you have three different calls to `fmt.Fprintf`, `fmt.Fprint`, and `fmt.Fprintln` all in quick succession when a simple `fmt.Fprinf(out, "POD LOGS" %s\n%s\n", h.Name, string(hookLog))` would've accomplished the same thing.

Reply:
The reason is because that's how the code was already written, as can be seen in the diff. I've only changed `io.Copy(out, logReader)` to `fmt.Fprint(out, string(hookLog))`.

Just to be clear, do you mean that `err` needs to be handled before `Fprintln` is called on line 116? If so, why was that not the case before (when the logs were written using `io.Copy` instead of `fmt.Fprint`)?

Reply:
Suggested change made in 73000df8dd16911783fd1e860a262cf6d0221c9b.